pkg/machine/applehv/machine.go

```diff
@@ -186,9 +187,40 @@ func (m *MacMachine) writeIgnitionConfigFile(opts machine.InitOptions, key strin
 
 func (m *MacMachine) Init(opts machine.InitOptions) (bool, error) {
 	var (
-		key          string
-		virtiofsMnts []machine.VirtIoFs
+		key                string
+		virtiofsMnts       []machine.VirtIoFs
+		cleanCallbackFuncs []func() error
+		err                error
 	)
+
+	// cleanup half-baked files if init fails at any point, or if init is sigterm'd
+	cleanCallback := func() {
+		for _, clean := range cleanCallbackFuncs {
```

Comment:
can this just go into the defer'd func?

Reply:
Yes, I only put it here since it's re-used in the signal logic portion